File: test/testspringboot/src/test/java/pers/acp/test/application/test/RestTest.java
package pers.acp.test.application.test;

import com.fasterxml.jackson.databind.JsonNode;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

/**
 * @author zhangbin by 28/04/2018 15:54
 * @since JDK 11
 */
class RestTest extends BaseTest {

    @Test
    void testGet() {
        String name = "abc";
        String pwd = "qwertyyu";
        ResponseEntity<JsonNode> result = testRestTemplate.getForEntity("/boot/rest2/" + name + "?pwd={1}", JsonNode.class, pwd);
        Assertions.assertEquals(HttpStatus.OK, result.getStatusCode());
        Assertions.assertEquals(result.getBody().get("result2").textValue(), name + pwd);
    }

    @Test
    void testPost() {
        String request = "sdfasdafsfa范德萨范德萨";
        ResponseEntity<JsonNode> result = testRestTemplate.postForEntity("/boot/rest2", request, JsonNode.class);
        Assertions.assertEquals(HttpStatus.OK, result.getStatusCode());
        Assertions.assertEquals(result.getBody().get("result2").textValue(), request);
    }

}

File: test/testkotlin/src/test/kotlin/pers/acp/kotlin/test/junit/RestTest.kt
package pers.acp.kotlin.test.junit

import com.test.kotlin.entity.TableOne
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.http.HttpStatus

/**
 * @author zhangbin by 28/04/2018 17:18
 * @since JDK 11
 */
class RestTest : BaseTest() {

    @Test
    fun testQuery200() {
        val name = "abc"
        val result = testRestTemplate.getForEntity("/query?name={1}", TableOne::class.java, name)
        Assertions.assertEquals(HttpStatus.OK, result.statusCode)
        Assertions.assertEquals(result.body!!.name, name)
    }

    @Test
    fun testAdd201() {
        val name = "fsdafa"
        val tableOne = TableOne().apply {
            this.name = name
            this.value = 101.0
        }
        val result = testRestTemplate.postForEntity("/add", tableOne, TableOne::class.java)
        Assertions.assertEquals(HttpStatus.CREATED, result.statusCode)
        Assertions.assertEquals(result.body!!.name, name)
    }

}